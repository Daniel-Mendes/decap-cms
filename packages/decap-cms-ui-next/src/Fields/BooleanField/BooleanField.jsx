import React from 'react';

import Field from '../../Field';
import Switch from '../../Switch';

function BooleanField({ name, onChange, children, value, ...props }) {
  return (
    <Field
      {...props}
      labelTarget={name}
      control
      onClick={() => (value ? onChange(false) : onChange(true))}
    >
      <Switch
        onChange={() => (value && onChange ? onChange(false) : onChange(true))}
        checked={value}
      />
      {children && children}
    </Field>
  );
}

export default BooleanField;
