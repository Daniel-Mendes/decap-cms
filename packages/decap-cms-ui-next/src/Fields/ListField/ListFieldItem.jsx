import React, { useState, forwardRef } from 'react';
import { useTranslate } from 'react-polyglot';
import styled from '@emotion/styled';

import { Button } from '../../Buttons';
import Tree from '../../Tree';
import { FieldContext } from '../../Field';
import { Menu, MenuItem, MenuSeparator } from '../../Menu';

const ListItem = styled.div`
  margin-left: -1rem;
  margin-right: -1rem;
  padding-left: 1rem;
  padding-right: 1rem;
  position: relative;
`;
const ListIconActions = styled.div`
  margin-right: -0.5rem;
`;
const AddNewHoverZone = styled.div`
  position: absolute;
  left: 0;
  right: 0;
  bottom: -0.625rem;
  display: flex;
  justify-content: center;
  align-items: center;
  z-index: 1;
`;
const AddNewIconButton = styled(Button)`
  line-height: 1;
  padding: 0 0.75rem;
  &,
  &:hover,
  &:focus,
  &:active:hover,
  &:focus:hover {
    background-color: ${({ theme }) => theme.color.background};
  }
  transition: 200ms;
  transform: scale(0);

  ${AddNewHoverZone}:hover & {
    transform: scale(1);
  }
`;

const ListFieldItem = forwardRef(
  (
    {
      itemExpanded,
      labelSingular,
      index,
      item,
      fields,
      onDelete,
      addListItem,
      moveListItem,
      handleChange,
      toggleExpand,
      last,
      className,
    },
    ref,
  ) => {
    const t = useTranslate();

    const [anchorEl, setAnchorEl] = useState(null);
    const [treeType, setTreeType] = useState(null);

    function handleOpenMenu(event) {
      setAnchorEl(event.currentTarget);
    }

    function handleClose() {
      setAnchorEl(null);
    }

    return (
      <ListItem className={className} ref={ref}>
        <Tree
          onExpandToggle={() => toggleExpand(index)}
          expanded={itemExpanded}
          label={labelSingular}
          description={!!Object.keys(item).length && item[Object.keys(item)[0]]}
          type={treeType}
          onHeaderMouseEnter={() => setTreeType('primary')}
          onHeaderMouseLeave={() => setTreeType(null)}
          actions={() => (
            <ListIconActions>
              <Button icon="more-horizontal" expanded={itemExpanded} onClick={handleOpenMenu} />
              <Button
                icon="trash-2"
                type="danger"
                expanded={itemExpanded}
                onMouseEnter={() => setTreeType('danger')}
                onMouseLeave={() => setTreeType(null)}
                onClick={() => onDelete(index)}
              />
              <Menu anchorEl={anchorEl} open={!!anchorEl} onClose={handleClose}>
                <MenuItem
                  icon="arrow-up"
                  onClick={() => {
                    moveListItem(index, index - 1);
                    handleClose();
                  }}
                  disabled={index === 0}
                >
                  {t('editor.editorWidgets.list.actions.moveUp')}
                </MenuItem>
                <MenuItem
                  icon="arrow-down"
                  onClick={() => {
                    moveListItem(index, index + 1);
                    handleClose();
                  }}
                  disabled={index === last}
                >
                  {t('editor.editorWidgets.list.actions.moveDown')}
                </MenuItem>

                <MenuSeparator />

                <MenuItem
                  icon="plus-circle"
                  onClick={() => {
                    addListItem(index);
                    handleClose();
                  }}
                >
                  {t('editor.editorWidgets.list.actions.addNewAbove')}
                </MenuItem>
                <MenuItem
                  icon="plus-circle"
                  onClick={() => {
                    addListItem(index + 1);
                    handleClose();
                  }}
                >
                  {t('editor.editorWidgets.list.actions.addNewBelow')}
                </MenuItem>

                <MenuSeparator />

                <MenuItem
                  icon="copy"
                  onClick={() => {
                    addListItem(index + 1, item);
                    handleClose();
                  }}
                  onMouseEnter={() => setTreeType('success')}
                  onMouseLeave={() => setTreeType(null)}
                >
                  {t('editor.editorWidgets.list.actions.duplicate')}
                </MenuItem>
                <MenuItem
                  icon="trash-2"
                  onClick={() => {
                    onDelete(index);
                    handleClose();
                  }}
                  onMouseEnter={() => setTreeType('danger')}
                  onMouseLeave={() => setTreeType(null)}
                  type="danger"
                >
                  {t('editor.editorWidgets.list.actions.delete')}
                </MenuItem>
              </Menu>
            </ListIconActions>
          )}
        >
          <FieldContext.Provider value={{ filled: true }}>
            {fields && fields(handleChange, index)}
          </FieldContext.Provider>
        </Tree>
        {!last && (
          <AddNewHoverZone>
            <AddNewIconButton
              icon="plus-circle"
              onClick={() => addListItem(index + 1)}
              iconSize="1.5rem"
            />
          </AddNewHoverZone>
        )}
      </ListItem>
    );
  },
);

ListFieldItem.displayName = 'ListFieldItem';

export default ListFieldItem;
