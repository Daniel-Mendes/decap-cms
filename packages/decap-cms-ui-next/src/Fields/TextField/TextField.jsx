import React, { useState, forwardRef } from 'react';
import styled from '@emotion/styled';

import Field from '../../Field';

export const StyledInput = styled.input`
  color: ${({ theme }) => theme.color.highEmphasis};
  background: none;
  border: none;
  outline: none;
  width: calc(100% + 32px);
  font-family: inherit;
  font-size: ${({ title }) => (title ? '2rem' : '1rem')};
  font-weight: ${({ title }) => (title ? 'bold' : 'normal')};
  letter-spacing: ${({ title }) => (title ? '-0.5px' : '0')};
  line-height: 1rem;
  caret-color: ${({ theme, error }) =>
    error ? theme.color.danger['900'] : theme.color.primary['800']};
  margin: -2rem -1rem -1rem -1rem;
  padding: 2rem 1rem 1rem 1rem;
  ${({ clickable }) => (clickable ? `cursor: pointer;` : ``)} ::placeholder {
    color: ${({ theme }) => theme.color.disabled};
  }
`;

const TextField = forwardRef(
  (
    {
      name,
      label,
      icon,
      description,
      status,
      onChange,
      onClick,
      readOnly,
      placeholder,
      children,
      value,
      focus,
      title,
      type,
      password,
      className,
      inline,
      error,
      errors,
    },
    ref,
  ) => {
    const [inputFocus, setInputFocus] = useState();

    return (
      <Field
        onClick={onClick}
        label={label}
        labelTarget={name}
        description={description}
        status={status}
        focus={focus || inputFocus}
        className={className}
        inline={inline}
        error={error}
        errors={errors}
        icon={icon}
        clickable={!readOnly}
      >
        <StyledInput
          ref={ref}
          clickable={readOnly && !!onClick}
          onClick={onClick}
          readOnly={readOnly}
          type={password ? 'password' : type ? type : 'text'}
          id={name}
          name={name}
          placeholder={placeholder ? placeholder : label ? `Type ${label.toLowerCase()} here` : ''}
          value={value || ''}
          onChange={e => onChange(e)}
          onFocus={() => setInputFocus(true)}
          onBlur={() => setInputFocus(false)}
          autoComplete="off"
          title={title}
          error={error}
        />
        {children && children}
      </Field>
    );
  },
);

TextField.displayName = 'TextField';

export default TextField;
