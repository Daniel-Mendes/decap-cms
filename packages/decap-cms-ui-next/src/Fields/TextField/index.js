export { default, StyledInput } from './TextField';
