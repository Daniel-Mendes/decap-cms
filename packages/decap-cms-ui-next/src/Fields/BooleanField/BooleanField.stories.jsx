import React from 'react';
import styled from '@emotion/styled';
import { useArgs } from '@storybook/preview-api';

import BooleanField from '.';

const StyledBooleanField = styled(BooleanField)`
  width: 33vw;
`;

export default {
  title: 'Fields/BooleanField',
  component: BooleanField,
  argTypes: {
    errors: {
      control: 'object',
      if: {
        arg: 'error',
      },
    },
  },
  args: {
    label: 'Boolean Field Label',
    description: 'This is a description',
    status: 'Required',
    value: false,
    inline: false,
    error: false,
    errors: [
      {
        message: 'Error message.',
      },
    ],
  },
};

export function _BooleanField(args) {
  const [{ value }, updateArgs] = useArgs();

  function toggleValue() {
    updateArgs({ value: !value });
  }

  return <StyledBooleanField {...args} value={value} onChange={toggleValue} />;
}
