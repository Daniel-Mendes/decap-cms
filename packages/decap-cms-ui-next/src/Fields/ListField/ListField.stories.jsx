import React, { useCallback } from 'react';
import isEqual from 'lodash/isEqual';
import styled from '@emotion/styled';
import { useArgs } from '@storybook/preview-api';

import ListField from '.';
import TextField from '../TextField';
import BooleanField from '../BooleanField';

const StyledListField = styled(ListField)`
  width: 33vw;
`;

export default {
  title: 'Fields/ListField',
  component: ListField,
  argTypes: {
    items: {
      control: 'object',
    },
  },
  args: {
    label: 'Links',
    labelSingular: 'Link',
    filled: false,
    items: [],
  },
};

export function _ListField(args) {
  const [{ items }, updateArgs] = useArgs();

  function handleChange(updatedItems) {
    updateArgs({ items: updatedItems });
  }

  return (
    <StyledListField
      {...args}
      name="featureLinks"
      onChange={handleChange}
      fields={(setListItemValue, itemIndex) => (
        <>
          <TextField
            name="featureLinkText"
            label="Text"
            value={items[itemIndex] && items[itemIndex].featureLinkText}
            onChange={featureLinkText => setListItemValue({ featureLinkText }, itemIndex)}
          />
          <TextField
            name="featureLinkPath"
            label="Path"
            value={items[itemIndex] && items[itemIndex].featureLinkPath}
            onChange={featureLinkPath => setListItemValue({ featureLinkPath }, itemIndex)}
          />
          <BooleanField
            name="newWindow"
            label="Open In New Window"
            value={items[itemIndex] && items[itemIndex].newWindow}
            onChange={newWindow => setListItemValue({ newWindow }, itemIndex)}
          />
        </>
      )}
    />
  );
}
