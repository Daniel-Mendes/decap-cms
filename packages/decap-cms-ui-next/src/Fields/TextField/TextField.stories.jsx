import React from 'react';
import styled from '@emotion/styled';
import { useArgs } from '@storybook/preview-api';

import TextField from '.';

const StyledTextField = styled(TextField)`
  width: 33vw;
`;

export default {
  title: 'Fields/TextField',
  component: TextField,
  argTypes: {
    errors: {
      control: 'object',
      if: {
        arg: 'error',
      },
    },
  },
  args: {
    label: 'Text Field Label',
    placeholder: 'Type something here',
    description: 'This is a description',
    status: 'Required',
    value: '',
    title: false,
    filled: false,
    error: false,
    errors: [
      {
        message: 'Error message.',
      },
    ],
  },
};

export function _TextField(args) {
  const [{ value }, updateArgs] = useArgs();

  function handleChange(e) {
    updateArgs({ value: e.target.value });
  }

  return <StyledTextField {...args} value={value} onChange={handleChange} />;
}
