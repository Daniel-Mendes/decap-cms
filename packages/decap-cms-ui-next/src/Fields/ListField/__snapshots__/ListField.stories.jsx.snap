// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Fields/ListField _ListField smoke-test 1`] = `
"<div class="e1y70o3o0 css-12auuey-FieldWrap-StyledListField e1x0ipyd6"><div class="css-t9hsmw-FieldInside-FieldInside e1x0ipyd5"><label for="featureLinks" class="css-r9gw8d-Label-StyledLabel e1x0ipyd3">Links </label><div class="css-dj548v-ChildrenWrap e1x0ipyd4"><div class="css-s978ft-ActionWrap eyaljbr1"></div><ul class="css-863xf9-StyledSortableContainer eyaljbr3"><div id="DndDescribedBy-0" style="display: none;">
    To pick up a draggable item, press the space bar.
    While dragging, use the arrow keys to move the item.
    Press space again to drop the item in its new position, or press escape to cancel.
  </div><div id="DndLiveRegion-0" role="status" aria-live="assertive" aria-atomic="true" style="position: fixed; width: 1px; height: 1px; margin: -1px; border: 0px; padding: 0px; overflow: hidden; clip: rect(0px, 0px, 0px, 0px); clip-path: inset(100%); white-space: nowrap;"></div></ul><button type="primary" class="eyaljbr2 css-rct8bj-StyledButton-StyledButton-StyledButton e9pg3mh2"><svg xmlns="http://www.w3.org/2000/svg" width="20" height="20" class="css-1yl0z6r-StyledIcon e9pg3mh0" viewBox="0 0 24 24" fill="none" stroke="currentColor" stroke-width="2" stroke-linecap="round" stroke-linejoin="round"><line x1="12" y1="5" x2="12" y2="19"></line><line x1="5" y1="12" x2="19" y2="12"></line></svg>Add New Link</button></div></div></div>"
`;
